keras_tuner/tuners/gridsearch.py

```diff
@@ -0,0 +1,218 @@
+        seed: Optional integer, the random seed.
+        hyperparameters: Optional `HyperParameters` instance. Can be used to
+            override (or register in advance) hyperparameters in the search
+            space.
+        tune_new_entries: Boolean, whether hyperparameter entries that are
+            requested by the hypermodel but that were not specified in
+            `hyperparameters` should be added to the search space, or not. If
+            not, then the default value for these parameters will be used.
+            Defaults to True.
+        allow_new_entries: Boolean, whether the hypermodel is allowed to
+            request hyperparameter entries not listed in `hyperparameters`.
+            Defaults to True.
+    """
+
+    def __init__(
+        self,
+        objective=None,
+        max_trials=None,
+        seed=None,
+        hyperparameters=None,
+        allow_new_entries=True,
+        tune_new_entries=True,
+    ):
+        super(GridSearchOracle, self).__init__(
+            objective=objective,
+            max_trials=max_trials,
+            hyperparameters=hyperparameters,
+            tune_new_entries=tune_new_entries,
+            allow_new_entries=allow_new_entries,
+            seed=seed,
+        )
+
+    def populate_space(self, trial_id):
+        """Fill the hyperparameter space with values.
+
+        Args:
+            trial_id: A string, the ID for this Trial.
+
+        Returns:
+            A dictionary with keys "values" and "status", where "values" is
+            a mapping of parameter names to suggested values, and "status"
+            is the TrialStatus that should be returned for this trial (one
+            of "RUNNING", "IDLE", or "STOPPED").
+        """
+        if len(self.start_order) > 0:
+            last_trial = self.trials[self.start_order[-1]]
+            last_values = last_trial.hyperparameters.values
+            values = self._get_next_combination(last_values)
+        else:
+            # Use all default values for the first trial.
+            values = {hp.name: hp.default for hp in self.get_space().space}
+        if values is None:
+            return {"status": trial_module.TrialStatus.STOPPED, "values": None}
+        return {"status": trial_module.TrialStatus.RUNNING, "values": values}
+
+    def _get_next_combination(self, values):
+        """Get the next value combination.
+
+        This function is called on the second and later trials, but not on the
+        first one. Even new hps appear during search, the all hps in
+        self.get_space() should have corresponding values in the `values` arg.
+        This is because `values` is from the last trial, whose values has been
+        updated during that trial, which contains all seen hps.
+
+        We treat the default value of a hp as the first value, and the rest are
+        in normal order. We will enumerate the values according to this order.
+        Get all possible hyperparameter values
```

Comment:
The description paragraphs above are unclear and have some typos, please clarify.